apps/fluent-tester/src/E2E/common/NativeTesting/pages/NativeTestingPageObject.win.ts

```diff
@@ -0,0 +1,45 @@
+import { BasePage, By } from '../../../common/BasePage.win';
+
+class NativeTestingPageObject extends BasePage {
+  waitForScrollViewDisplayed(timeout?: number): void {
+    browser.waitUntil(
+      () => {
+        return this.doesScrollViewParentExist();
+      },
+      {
+        timeout: timeout ?? this.waitForPageTimeout,
+        timeoutMsg:
+          'The ScrollView containing all test page buttons was not found. Did you remove the ScrollView or change the testID? The testID must stay constant to align with our native code.',
+        interval: 1000,
+      },
+    );
+  }
+
+  doesScrollViewParentExist(): boolean {
+    return this._scrollViewParent.isExisting();
+  }
+
+  validateScrollViewChildren() {
+    // Gets all the children
+    const testChildren = this._scrollViewParent.$$('//*');
+
+    // All the button children have the same format. We want to find these buttons
+    const reg = new RegExp('Homepage_[a-zA-Z]*_Button');
+    const buttonList = [];
+
+    testChildren.forEach((child) => {
+      const autoId = child.getAttribute('AutomationId');
```

Comment:
Dedupe the getAttribute call, but also this fn will return true if any of the buttons match, not all of the buttons match, right?  I think we should be reversing the two conditions e.g.

`forEach {
...
if (!autoId || !autoId.match(reg))   { return false; }
...
}`

`return testChildren.length > 0; // and still check that we _do_ have buttons matching the format`

Reply:
sidebar wondering for our test in this repo that opens all of the test pages if we should have a mechanism for verifying that clicking each button child actually generated the expected page change... but maybe that's unnecessary red tape

Reply:
You're right. Initially, I wanted to see if the regex worked and output all the contents of the array, but I forgot to change it back. However, I'm changing it to something a bit different... If one button matches the regex, we return true immediately.


In regards to your second statement, if I'm understanding correctly, we do already have that (to an extent). When clicking on each button, we wait for the page load and grab a UI element specific to that page to validate the correct content is being loaded.

Reply:
> If one button matches the regex, we return true immediately.

Why wouldn't you check all the buttons?

Reply:
It's a good question. But what would the success/failure condition be like?

If we want to validate every single component, then we have to hardcode a component list somewhere. And that list will have to be upkept when new controls are made, or name changes, etc.

I think the goal here is to simply check that the **UI Structure** is maintained. So not checking every individual component, just making sure no engineer changes the UI structure, because that would mess up native

Reply:
I see, that makes sense, and we could keep it as is, but shouldn't we also be checking that each button's automationID matches the expected format? I don't know if there's any other safeguards in place for that. I'm just wondering how we could catch those mistakes to ensure we don't unintentionally remove or exclude a test page from being tested if the format is wrong.

I understand

Reply:
I do think that's a good idea. That'll make it so no one changes those testID values as well. Changed in my newest commit